sdk/servicebus/service-bus/src/receivers/receiverCommon.ts

```diff
@@ -155,28 +168,47 @@ export function deadLetterMessage(
   const dispositionStatusOptions: DispositionStatusOptions = {
     propertiesToModify: actualPropertiesToModify,
     deadLetterReason: propertiesToModify?.deadLetterReason,
-    deadLetterDescription: propertiesToModify?.deadLetterErrorDescription
+    deadLetterDescription: propertiesToModify?.deadLetterErrorDescription,
+    retryOptions
   };
 
-  return settleMessage(
-    message,
-    DispositionType.deadletter,
-    context,
-    entityPath,
-    dispositionStatusOptions
-  );
+  return settleMessage(message, DispositionType.deadletter, context, entityPath, {
+    ...dispositionStatusOptions,
+    retryOptions
+  });
 }
 
 /**
  * @internal
- *
  */
-function settleMessage(
+export function settleMessage(
   message: ServiceBusMessageImpl,
   operation: DispositionType,
   context: ConnectionContext,
   entityPath: string,
-  options?: DispositionStatusOptions
+  options: DispositionStatusOptions,
+  _settleMessageOperation: typeof settleMessageOperation = settleMessageOperation
+): Promise<void> {
+  return retry({
```

Comment:
Wow, did not realize settling messages wasn't retried before.

Reply:
There is an issue for it!! https://github.com/Azure/azure-sdk-for-js/issues/2885